src/Microsoft.AspNet.WebUtilities/KeyValueAccumulator.cs

```diff
@@ -7,17 +7,16 @@
 
 namespace Microsoft.AspNet.WebUtilities
 {
-    public class KeyValueAccumulator
+    public struct KeyValueAccumulator
```

Comment:
Does this even need an inner dictionary? It could be just have a callback OnKeyValuePair(string key, string value) or something.